sdk/provisioning/Azure.Provisioning/src/Resource.cs

```diff
@@ -59,6 +61,10 @@ internal void AddDependency(Resource resource)
         /// </summary>
         public IList<Parameter> Parameters { get; }
 
+        internal IList<Output> Outputs { get; }
+
+        internal IConstruct? ModuleScope { get; set; }
```

Comment:
Would `SynthesizedModuleScope` be more clear that this is not an intrinsic property of the resource?

Did you consider a SynthesizedResource which takes a resource and has this property on it?

Reply:
Filed https://github.com/Azure/azure-sdk-for-net/issues/42035